_docs/4-kubernetes/3-bookinfo.md

```diff
@@ -0,0 +1,200 @@
+
+   Confirm that all services and pods are correctly defined and running:
+
+   ```bash
+   $ kubectl get services
+   NAME          CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
+   details       None            <none>        9080/TCP         1m
+   gateway       10.102.224.69   <nodes>       6379:32000/TCP   1m
+   kubernetes    10.96.0.1       <none>        443/TCP          22d
+   productpage   None            <none>        9080/TCP         1m
+   ratings       None            <none>        9080/TCP         1m
+   reviews       None            <none>        9080/TCP         1m
+   ```
+
+   and
+
+   ```bash
+   $ kubectl get pods
+   NAME                   READY     STATUS    RESTARTS   AGE
+   details-v1-w16wg       1/1       Running   0          1m
+   gateway-j6vr6          1/1       Running   0          1m
+   productpage-v1-chmvt   2/2       Running   0          1m
+   ratings-v1-2p06t       1/1       Running   0          1m
+   reviews-v1-v39cg       1/1       Running   0          1m
+   reviews-v2-3xk9f       2/2       Running   0          1m
+   reviews-v3-kcwpp       2/2       Running   0          1m
+   rules-56tf8            1/1       Running   0          18h
+   ```
+
+
+1. Expose the Gateway service
+
+   Determine the node on which the `gateway` pod runs, and use the node's IP address as the external gateway IP.
+
+   ```bash
+   $ kubectl describe pod gateway-j6vr6 | grep Node
+   Node:		node4.k8s.example.com/9.147.221.77
+   $ export GATEWAY_URL=9.147.221.77:32000
+   ```
+
+### Content Based Routing
+
+1. Set the default routes to v1 of each service
+
+   ```bash
+   $ kubectl create -f examples/k8s-bookinfo-default-routes.yaml
+   routingrule "set-productpage-v1" created
+   routingrule "set-details-v1" created
+   routingrule "set-ratings-v1" created
+   routingrule "set-reviews-v1" created
+   ```
+
+   You can confirm the routes are defined and valid by inspecting the rules' `Status` field returned in:
+
+   ```bash
+   $ kubectl get routingrules -o json
+   ```
+
+   If you open the Bookinfo URL (`http://$GATEWAY_URL/productpage/productpage`) in your browser,
+   you should see the bookinfo application `productpage` displayed. Notice that the `productpage`
+   is displayed, with no rating stars since `reviews:v1` does not access the ratings service.
+
+1. Route a specific user to `reviews:v2`
+
+   Lets enable the ratings service for test user "jason" by routing productpage traffic to
+   `reviews:v2` instances.
+
+   ```bash
+   $ kubectl create -f examples/k8s-bookinfo-jason-reviews-v2-route-rules.yaml
+   routingrule "set-reviews-v2-user-jason" created
+   ```
+
+   Confirm the rule is applied and valid:
+
+   ```bash
+   $ kubectl get routingrule set-reviews-v2-user-jason -o json
+   {
+      "apiVersion": "amalgam8.io/v1",
+      "kind": "RoutingRule",
+      ...
+      "status": {
+         "state": "valid"
+      }
```

Comment:
No, just that they confirm to the expected syntax and that they **can** be pulled by the sidecars.